persistence/trees/trees.go

```diff
@@ -88,6 +99,18 @@ type treeStore struct {
 	treeStoreDir string
 	rootTree     *stateTree
 	merkleTrees  map[string]*stateTree
+
+	// PrevState holds a previous view of the Worldstate.
+	// The tree store rolls back to this view if errors are encountered during block application.
+	PrevState *worldstate
+}
+
+// worldstate holds a (de)serializable view of the entire tree state.
+// TECHDEBT(#566) - Hook this up to node CLI subcommands
+type worldstate struct {
+	TreeStoreDir string
+	RootTree     *stateTree
+	MerkleTrees  map[string]*stateTree
```

Comment:
Apologize if I already brought this up in a previous PR but why do all of these need to be exposed outside this package? If we can change it, let's do it. If it's not, and I'm re-asking the same question, might mean it's an opportunity to add a comment with reasoning.


```suggestion
	prevState *worldstate
}

// worldstate holds a (de)serializable view of the entire tree state.
// TECHDEBT(#566) - Hook this up to node CLI subcommands
type worldstate struct {
	treeStoreDir string
	rootTree     *stateTree
	merkleTrees  map[string]*stateTree
```